Resync `resources` from WPT Upstream

Ahmad-S792 opened the pull request:
#### af0a52765978395b82ef69f634cfcda5d6b21548
<pre>
Resync `resources` from WPT Upstream
<a href="https://bugs.webkit.org/show_bug.cgi?id=306173">https://bugs.webkit.org/show_bug.cgi?id=306173</a>
<a href="https://rdar.apple.com/168822984">rdar://168822984</a>

Reviewed by NOBODY (OOPS!).

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/d61231d9ceef372cc66d64ac19c467c2506a8814">https://github.com/web-platform-tests/wpt/commit/d61231d9ceef372cc66d64ac19c467c2506a8814</a>

* LayoutTests/imported/w3c/web-platform-tests/resources/example.pdf: Added.
* LayoutTests/imported/w3c/web-platform-tests/resources/idlharness.js:
(IdlInterface.prototype.default_to_json_operation):
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tox.ini:
* LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-actions.js:
(Actions.prototype.addTick):
(PointerSource.prototype.pointerMove):
(WheelSource.prototype.scroll):
* LayoutTests/imported/w3c/web-platform-tests/resources/testdriver.js:
(assertTestIsTentative):
(window.test_driver.bidi.):
(window.test_driver):
* LayoutTests/imported/w3c/web-platform-tests/resources/testharness.js:
(promise_rejects_quotaexceedederror):
* LayoutTests/imported/w3c/web-platform-tests/resources/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/resources/web-extensions-helper.js: Added.
(globalThis.runTestsWithWebExtension):
* LayoutTests/imported/w3c/web-platform-tests/resources/webidl2/lib/VERSION.md:
* LayoutTests/imported/w3c/web-platform-tests/resources/webidl2/lib/webidl2.js:
(return.):
* LayoutTests/tests-options.json:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/af0a52765978395b82ef69f634cfcda5d6b21548

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/153142 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/25924 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/19522 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/161886 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/106600 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/44ea0017-510e-4caa-9e8e-d83e3b673800) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/26451 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/26229 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/118381 "Found 19 new test failures: http/wpt/cache-storage/cache-quota-add.any.html imported/w3c/web-platform-tests/WebCryptoAPI/getRandomValues.any.html imported/w3c/web-platform-tests/WebCryptoAPI/getRandomValues.any.worker.html imported/w3c/web-platform-tests/compression/idlharness.https.any.html imported/w3c/web-platform-tests/compression/idlharness.https.any.worker.html imported/w3c/web-platform-tests/encoding/idlharness.any.html imported/w3c/web-platform-tests/encoding/idlharness.any.serviceworker.html imported/w3c/web-platform-tests/encoding/idlharness.any.sharedworker.html imported/w3c/web-platform-tests/encoding/idlharness.any.worker.html imported/w3c/web-platform-tests/html/infrastructure/safe-passing-of-structured-data/structured-cloning-error-stack-optional.sub.window.html ... (failure)") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/83826 "6 flakes 2 failures") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/f55864a1-0471-476e-ac8f-3d13039e03f8) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/156101 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/20622 "Found 1 new test failure: http/wpt/cache-storage/cache-quota-add.any.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/137480 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/99094 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/74b385c5-481c-48b7-9942-734350ddca26) 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/19696 "Found 22 new test failures: imported/w3c/web-platform-tests/WebCryptoAPI/getRandomValues.any.html imported/w3c/web-platform-tests/WebCryptoAPI/getRandomValues.any.worker.html imported/w3c/web-platform-tests/compression/idlharness.https.any.html imported/w3c/web-platform-tests/compression/idlharness.https.any.worker.html imported/w3c/web-platform-tests/encoding/idlharness.any.html imported/w3c/web-platform-tests/encoding/idlharness.any.serviceworker.html imported/w3c/web-platform-tests/encoding/idlharness.any.sharedworker.html imported/w3c/web-platform-tests/encoding/idlharness.any.worker.html imported/w3c/web-platform-tests/file-system-access/idlharness.https.any.html imported/w3c/web-platform-tests/file-system-access/idlharness.https.any.worker.html ... (failure)") | [❌ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/17638 "2 api tests failed or timed out") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/9722 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/129348 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/15353 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/164360 "Built successfully") | | 
| | | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/16947 "Found 20 new test failures: http/wpt/cache-storage/cache-quota-add.any.html http/wpt/cache-storage/cache-quota-after-restart.any.html imported/w3c/web-platform-tests/WebCryptoAPI/getRandomValues.any.html imported/w3c/web-platform-tests/WebCryptoAPI/getRandomValues.any.worker.html imported/w3c/web-platform-tests/compression/idlharness.https.any.html imported/w3c/web-platform-tests/compression/idlharness.https.any.worker.html imported/w3c/web-platform-tests/file-system-access/idlharness.https.any.html imported/w3c/web-platform-tests/file-system-access/idlharness.https.any.worker.html imported/w3c/web-platform-tests/fs/idlharness.https.any.html imported/w3c/web-platform-tests/fs/idlharness.https.any.worker.html ... (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/126441 "Found 19 new test failures: http/wpt/cache-storage/cache-quota-add.any.html imported/w3c/web-platform-tests/WebCryptoAPI/getRandomValues.any.html imported/w3c/web-platform-tests/WebCryptoAPI/getRandomValues.any.worker.html imported/w3c/web-platform-tests/compression/idlharness.https.any.html imported/w3c/web-platform-tests/compression/idlharness.https.any.worker.html imported/w3c/web-platform-tests/encoding/idlharness.any.html imported/w3c/web-platform-tests/encoding/idlharness.any.serviceworker.html imported/w3c/web-platform-tests/encoding/idlharness.any.sharedworker.html imported/w3c/web-platform-tests/encoding/idlharness.any.worker.html imported/w3c/web-platform-tests/html/infrastructure/safe-passing-of-structured-data/structured-cloning-error-stack-optional.sub.window.html ... (failure)") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/25721 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/21673 "Found 24 new test failures: http/wpt/cache-storage/cache-quota-add.any.html http/wpt/cache-storage/cache-quota-after-restart.any.html imported/w3c/web-platform-tests/WebCryptoAPI/getRandomValues.any.html imported/w3c/web-platform-tests/WebCryptoAPI/getRandomValues.any.worker.html imported/w3c/web-platform-tests/compression/idlharness.https.any.html imported/w3c/web-platform-tests/compression/idlharness.https.any.worker.html imported/w3c/web-platform-tests/encoding/idlharness.any.html imported/w3c/web-platform-tests/encoding/idlharness.any.serviceworker.html imported/w3c/web-platform-tests/encoding/idlharness.any.sharedworker.html imported/w3c/web-platform-tests/encoding/idlharness.any.worker.html ... (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/126599 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/25723 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/137149 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/82386 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/21551 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/13928 "Found 20 new test failures: http/wpt/cache-storage/cache-quota-add.any.html http/wpt/cache-storage/cache-quota-after-restart.any.html imported/w3c/web-platform-tests/WebCryptoAPI/getRandomValues.any.html imported/w3c/web-platform-tests/WebCryptoAPI/getRandomValues.any.worker.html imported/w3c/web-platform-tests/compression/idlharness.https.any.html imported/w3c/web-platform-tests/compression/idlharness.https.any.worker.html imported/w3c/web-platform-tests/file-system-access/idlharness.https.any.html imported/w3c/web-platform-tests/file-system-access/idlharness.https.any.worker.html imported/w3c/web-platform-tests/fs/idlharness.https.any.html imported/w3c/web-platform-tests/fs/idlharness.https.any.worker.html ... (failure)") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/25339 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/25032 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/25190 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/25091 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (commented):
We should try to implement https://github.com/whatwg/webidl/pull/1465 if we do decide to sync this, to avoid losing test coverage

Ahmad-S792 committed:
Resync `resources` from WPT Upstream
https://bugs.webkit.org/show_bug.cgi?id=306173
rdar://168822984

Reviewed by NOBODY (OOPS!).

Upstream commit: https://github.com/web-platform-tests/wpt/commit/d61231d9ceef372cc66d64ac19c467c2506a8814

* LayoutTests/imported/w3c/web-platform-tests/resources/example.pdf: Added.
* LayoutTests/imported/w3c/web-platform-tests/resources/idlharness.js:
(IdlInterface.prototype.default_to_json_operation):
* LayoutTests/imported/w3c/web-platform-tests/resources/test/tox.ini:
* LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-actions.js:
(Actions.prototype.addTick):
(PointerSource.prototype.pointerMove):
(WheelSource.prototype.scroll):
* LayoutTests/imported/w3c/web-platform-tests/resources/testdriver.js:
(assertTestIsTentative):
(window.test_driver.bidi.):
(window.test_driver):
* LayoutTests/imported/w3c/web-platform-tests/resources/testharness.js:
(promise_rejects_quotaexceedederror):
* LayoutTests/imported/w3c/web-platform-tests/resources/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/resources/web-extensions-helper.js: Added.
(globalThis.runTestsWithWebExtension):
* LayoutTests/imported/w3c/web-platform-tests/resources/webidl2/lib/VERSION.md:
* LayoutTests/imported/w3c/web-platform-tests/resources/webidl2/lib/webidl2.js:
(return.):
* LayoutTests/tests-options.json:
